DisCatSharp.HybridCommands/HybridCommandsExtension.cs

```diff
@@ -0,0 +1,273 @@
+// This file is part of the DisCatSharp project, based off DSharpPlus.
+//
+// Copyright (c) 2023 AITSYS
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+using System;
+
+using Microsoft.Extensions.Logging;
+using DisCatSharp.ApplicationCommands;
+using DisCatSharp.CommandsNext;
+using System.Reflection;
+using System.Linq;
+using DisCatSharp.ApplicationCommands.Context;
+using System.Threading.Tasks;
+using DisCatSharp.HybridCommands.Context;
+using DisCatSharp.HybridCommands.Attributes;
+using System.Collections.Generic;
+using DisCatSharp.Enums;
+using DisCatSharp.Entities;
+using System.Diagnostics;
+using DisCatSharp.EventArgs;
+
+namespace DisCatSharp.HybridCommands;
+public sealed class HybridCommandsExtension : BaseExtension
+{
+	internal static HybridCommandsConfiguration? Configuration;
+
+	internal static ILogger? Logger { get; set; }
+
+	internal static string? ExecutionDirectory { get; set; }
+
+	internal List<HybridCommandAttribute> registeredCommands { get; set; } = new();
+
+	public static bool DebugEnabled 
+		=> Configuration?.DebugEnabled ?? false;
+
+	internal protected override async void Setup(DiscordClient client)
+	{
+		if (this.Client != null)
+			throw new InvalidOperationException("What did I tell you?");
```

Comment:
what is this error supposed to mean?

Reply:
it's just missing all docs...... @TheXorog -.-

Reply:
/// <summary>
	/// Runs setup.
	/// <note type="caution">DO NOT RUN THIS MANUALLY. DO NOT DO ANYTHING WITH THIS.</note>
	/// </summary>
	/// <param name="client">The client to setup on.</param>

Reply:
oh nooooo its missing documentation on an internal method we know what it does, oh so terrible

Reply:
Docs are not yet a priority, the setup method i copied from another project and modified it to suit my needs

Reply:
> oh nooooo its missing documentation on an internal method we know what it does, oh so terrible

Mira. I said it often enough that I aim to document everything so others can work on it as well.
Docs have the same prio as the implementation.